distributed/worker_state_machine.py

```diff
@@ -1062,7 +1061,7 @@ def __init__(
         self.has_what = defaultdict(set)
         self.data_needed = HeapSet(key=operator.attrgetter("priority"))
         self.data_needed_per_worker = defaultdict(
-            lambda: HeapSet(key=operator.attrgetter("priority"))
+            cast(Callable, partial(HeapSet, key=operator.attrgetter("priority")))
```

Comment:
this should work without a cast https://mypy-play.net/?mypy=latest&python=3.10&flags=strict&gist=e7185bc44629717e7af27fe9d359a046

```suggestion
            partial(HeapSet[TaskState], key=operator.attrgetter("priority"))
```